frontend/src/components/common/WeeklyEventItemPopUp.tsx

```diff
@@ -86,35 +104,57 @@ const WeeklyEventItemPopUp = ({
                   </HStack>
                   <HStack py="0.5rem">
                     <TimeIcon />
-                    <Text textStyle="desktopSmall">{convertTime(schedule!.startTime)} - {convertTime(schedule!.endTime)}</Text>
+                    <Text textStyle={isMobile ? "mobileSmall" : "desktopSmall"}>
+                      {convertTime(schedule!.startTime)} -{" "}
+                      {convertTime(schedule!.endTime)}
+                    </Text>
                   </HStack>
-                  {
-                    nextDropOff && (
-                      <Text textStyle="desktopSmall" py="0.5rem">
-                        Next Dropoff: {new Date(nextDropOff).toLocaleString(undefined, {
-                          year: "numeric",
-                          month: "long",
-                          day: "numeric",
-                        })}
-                      </Text>
-                    )
-                  }
+                  {nextDropOff && (
+                    <Text textStyle={isMobile ? "mobileSmall" : "desktopSmall"} py="0.5rem">
+                      Next Dropoff:{" "}
+                      {new Date(nextDropOff).toLocaleString(undefined, {
+                        year: "numeric",
+                        month: "long",
+                        day: "numeric",
+                      })}
+                    </Text>
+                  )}
+                  {isMobile && (
+                    <Badge

```

Comment:
Wondering if Badge should be a component...

Reply:
it can, you can create a variant for it similar to what we have for the buttons in `theme.tsx`